packages/components/organisms/f-checkout/src/store/analytics.module.js

```diff
@@ -0,0 +1,131 @@
+                commit(UPDATE_AUTOFILL, JSON.parse(JSON.stringify(payload)));
+            }
+
+            commit(UPDATE_ANALYTICS_STATE, checkoutState);
+        },
+
+        /**
+         * Pushes initial state of checkout to the dataLayer.
+         */
+        updateChangedFields ({ commit }, field) {
+            commit(UPDATE_CHANGED_FIELDS, field);
+        },
+
+        /**
+         * Pushes initial state of checkout to the dataLayer.
+         */
+        trackInitialLoad ({ state, dispatch }) {
+            window.dataLayer = window.dataLayer || [];
+
+            const pageName = state.isLoggedIn ? 'Overview' : 'Guest';
+
+            Trak.event({
+                custom: {
+                    checkout: {
+                        step: 1
+                    },
+                    basket: {
+                        id: state.basket.id,
+                        total: state.basket.total
+                    },
+                    restaurant: {
+                        id: state.restaurantId
+                    },
+                    pageData: {
+                        name: `Checkout 1 ${pageName}`,
+                        group: 'Checkout'
+                    }
+                }
+            });
+
+            dispatch('trackFormInteraction', { action: 'start' });
+        },
+
+        /**
+         * Pushes `form` event to the dataLayer with correct data
+         */
+        trackFormInteraction ({ state }, { action, error }) {
+            const formName = state.isLoggedIn ? 'checkout' : 'checkout_guest';
+
+            Trak.event({
+                event: 'Form',
+                custom: {
+                    form: {
+                        name: formName,
+                        action,
+                        error: mapAnalyticsFieldNames(error) || null,
+                        autofill: JSON.parse(JSON.stringify(state.autofill)),
+                        changes: JSON.parse(JSON.stringify(state.changedFields))
+                    }
+                }
+            });
+        }
+    },
+
+    mutations: {
+        [UPDATE_ANALYTICS_STATE]: (state, checkoutState) => {
+            state.isLoggedIn = checkoutState.isLoggedIn || null;
+            state.basket = checkoutState.basket || null;
+            state.restaurantId = checkoutState.restaurantId || null;
+            state.serviceType = checkoutState.serviceType || null;
+        },
+
+        [UPDATE_CHANGED_FIELDS]: (state, field) => {
+            const analyticsNames = mapAnalyticsFieldNames(field);
+
+            if (!state.changedFields.includes(analyticsNames)) {
+                state.changedFields.push(analyticsNames);
+            }
+        },
+
+        [UPDATE_AUTOFILL]: (state, payload) => {
+            const { customer, address, serviceType } = payload;
+
+            if (customer.mobileNumber) {
+                state.autofill.push('phone');
+            }
+
+            if (serviceType === 'delivery') {
```

Comment:
Do we need this conditional at all? We have `if (value)` below, so I assume in the case of collection we won't have a value, and most importantly, probably `address` will be `[]`, so there won't be a `forEach`. I might be wrong.

Reply:
If you do remove the conditional because of this, I'd probably add a comment just saying that it's only relevant for delivery orders and why (as it's info I wouldn't know without context of the data being passed in)

Reply:
Thanks, I've added a comment to this.
@SachaZvetelman I was hoping the checkout state for address would be `{ address: { line1: '', line2: '' ....}` because that's what the default state allowed?